src/main/java/frc/trigon/robot/subsystems/elevator/Elevator.java

```diff
@@ -0,0 +1,68 @@
+package frc.trigon.robot.subsystems.elevator;
+
+import edu.wpi.first.units.Units;
+import edu.wpi.first.wpilibj.sysid.SysIdRoutineLog;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import frc.trigon.robot.subsystems.MotorSubsystem;
+import org.littletonrobotics.junction.Logger;
+
+public class Elevator extends MotorSubsystem {
+    private final static Elevator INSTANCE = new Elevator();
+    private final ElevatorIO elevatorIO = ElevatorIO.generateIO();
+    private final ElevatorInputsAutoLogged elevatorInputs = new ElevatorInputsAutoLogged();
+    private ElevatorConstants.ElevatorState targetState = ElevatorConstants.ElevatorState.BOTTOM;
+
+    public static Elevator getInstance() {
+        return INSTANCE;
+    }
+
+    private Elevator() {
+        setName("Elevator");
+    }
+
+    @Override
+    public void periodic() {
+        elevatorIO.updateInputs(elevatorInputs);
+        Logger.processInputs("Elevator", elevatorInputs);
+        updateMechanism();
+    }
+
+    @Override
+    public void updateLog(SysIdRoutineLog log) {
+        log.motor("Elevator")
+                .linearPosition(Units.Meters.of(elevatorInputs.positionMeters))
+                .linearVelocity(Units.MetersPerSecond.of(elevatorInputs.velocityMetersPerSecond))
+                .voltage(Units.Volts.of(elevatorInputs.motorVoltage));
+    }
+
+    @Override
+    public SysIdRoutine.Config getSysIdConfig() {
+        return ElevatorConstants.SYSID_CONFIG;
+    }
+
+    @Override
+    public void setBrake(boolean brake) {
+        elevatorIO.setBrake(brake);
+    }
+    
+    @Override
+    public void stop() {
+        elevatorIO.stop();
+    }
+
+    void setTargetState(ElevatorConstants.ElevatorState targetState) {
+        this.targetState = targetState;
+
+        elevatorIO.setTargetPosition(targetState.positionMeters);
+    }
+
+    private boolean atTargetState(ElevatorConstants.ElevatorState targetState) {
+        return this.targetState == targetState;
+    }
```

Comment:
1. this logic checks if the current target state is the one given by the param. I want this to check if we're _at_ the target position. (if the current position is the target one within a tolerance)
2. this should be public